source/fuzz/fuzzer_pass_add_access_chains.cpp

```diff
@@ -162,9 +168,25 @@ void FuzzerPassAddAccessChains::Apply() {
         // Apply the transformation to add an access chain.
         ApplyTransformation(TransformationAccessChain(
             GetFuzzerContext()->GetFreshId(), chosen_pointer->result_id(),
-            index_ids, instruction_descriptor));
+            index_ids, instruction_descriptor, fresh_ids_for_clamping));
       });
 }
 
+uint32_t FuzzerPassAddAccessChains::FindOrCreateIntegerConstantReadyForClamping(
+    uint32_t value, uint32_t bound,
+    std::vector<std::pair<uint32_t, uint32_t>>* fresh_ids_for_clamping) {
+  // Choose the signedness of the constant
+  bool is_signed = GetFuzzerContext()->ChooseEven();
```

Comment:
When you make this inline can you also avoid this temporary and use the "ChooseEven" experession as the actual parameter for FindOrCreateIntegerConstant?

Reply:
I think this would be a problem, because the constant of bounds (bound - 1) needs to have the same type as the index constant (and so the same signedness), so we need a way to make sure that the two are the same (the temporary is used twice)

Reply:
Oh - I didn't spot that it was used twice; please ignore the comment then.